libraries/core_libs/consensus/src/pbft/pbft_manager.cpp

```diff
@@ -1726,6 +1730,24 @@ bool PbftManager::pushPbftBlock_(PeriodData &&period_data, std::vector<std::shar
   return true;
 }
 
+void PbftManager::calculateBonusVoteCount(PeriodData &period_data) const {
+  // First period does not have any reward votes
+  if (period_data.pbft_blk->getPeriod() < 2) [[unlikely]] {
+    return;
+  }
+  const auto dpos_votes_count = final_chain_->dpos_eligible_total_vote_count(period_data.pbft_blk->getPeriod() - 1);
+  const auto min_weight = std::min<size_t>(COMMITTEE_SIZE, dpos_votes_count) * 2 / 3 + 1;
+  size_t total_weight = 0;
+
+  for (size_t i = 0; i < period_data.previous_block_cert_votes.size(); i++) {
+    assert(period_data.previous_block_cert_votes[i]->getWeight());
+    total_weight += *period_data.previous_block_cert_votes[i]->getWeight();
+    if (total_weight > min_weight) {
```

Comment:
if dpos_votes_count is lets say 2000 and I have 1600 so my vote would be enough, this condition would count it as bonus vote, which is not true. Overall this for loop with indexes could be probably more simple:
```

bool is_bonus_vote = false;
for (const auto&  vote: period_data.previous_block_cert_votes) {
  ...
 
  // need this delay to not count first vote > min_weight as bonus vote
  if (is bonus_vote) {
    period_data.bonus_votes_count++;
  }
  
  if (total_weight > min_weight) { 
    is_bonus_vote = true;
  }
}
```

Reply:
but this way you need to iterate whole array I do not need to. And yes, if I change order of line 1744 and put it under if, I would fix the case you mentioned

Reply:
yes, there should be break